### PR TITLE
[REEF-2035] BlobFs parameters should be account name and key

### PR DESCRIPTION
**Pull request by mandyshieh**
Took out the connection string parameter and replaced with account name
and account key. Connection string is then constructed whenever
necessary.

[REEF-2035] (https://issues.apache.org/jira/browse/REEF-2035)

Pull Request:
  This closes #

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/Parameters/AzureBlobStorageAccountKey.cs [hunk 1]
```diff
@@ -19,10 +19,10 @@
 
 namespace Org.Apache.REEF.IO.FileSystem.AzureBlob.Parameters
 {
-    [NamedParameter("The connection string for Azure Storage")]
-    public sealed class AzureStorageConnectionString : Name<string>
+    [NamedParameter("The Azure Blob Storage account key")]
+    public sealed class AzureBlobStorageAccountKey : Name<string>
```
Comment (tyclintw) [inline lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/Parameters/AzureBlobStorageAccountKey.cs]: >AzureBlobStorageAccountKey [](start = 24, length = 26)

FYI - This is an API surface breaking change. I much prefer this than the connection string. However, the REEF team may want us to still support the legacy ConnectionString parameter so be prepared to support both if necessary.

**Comment by homezcx** [comment]
@mandyshieh This PR needs rebase to resolve conflicts.

File: lang/cs/Org.Apache.REEF.Examples.HelloREEF/HelloREEF.cs [hunk 2]
```diff
@@ -104,16 +104,18 @@ private static IConfiguration GetRuntimeConfiguration(string name)
...
+                    const string blobStorageAccountName = "name";
+                    const string blobStorageAccountKey = "key";
+                    const string containerName = "foo";
                     return HDInsightClientConfiguration.ConfigurationModule
                         .Set(HDInsightClientConfiguration.HDInsightPasswordParameter, @"pwd")
                         .Set(HDInsightClientConfiguration.HDInsightUsernameParameter, @"foo")
                         .Set(HDInsightClientConfiguration.HDInsightUrlParameter, @"https://foo.azurehdinsight.net/")
-                        .Set(HDInsightClientConfiguration.JobSubmissionDirectoryPrefix, string.Format(@"/{0}/tmp", continerName))
-                        .Set(AzureBlockBlobFileSystemConfiguration.ConnectionString, connectionString)
+                        .Set(HDInsightClientConfiguration.JobSubmissionDirectoryPrefix, string.Format(@"/{0}/tmp", containerName))
```
Comment (motus) [inline lang/cs/Org.Apache.REEF.Examples.HelloREEF/HelloREEF.cs]: why not `$"/{containerName}/tmp"`?

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlobFileSystem.cs [hunk 3]
```diff
@@ -179,13 +179,14 @@ private sealed class TestContext
 
             public IFileSystem GetAzureFileSystem()
             {
-                var conf = AzureBlockBlobFileSystemConfiguration.ConfigurationModule
-                    .Set(AzureBlockBlobFileSystemConfiguration.ConnectionString, "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;")
+                var conf = AzureBlobFileSystemConfiguration.ConfigurationModule
+                    .Set(AzureBlobFileSystemConfiguration.AccountName, "devstoreaccount1")
+                    .Set(AzureBlobFileSystemConfiguration.AccountKey, "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==")
```
Comment (motus) [inline lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlobFileSystem.cs]: hmm, is that a real key published in open source?

File: lang/cs/Org.Apache.REEF.Examples.HelloREEF/HelloREEF.cs (hunk 2 above)
Comment (mandyshieh) [reply lang/cs/Org.Apache.REEF.Examples.HelloREEF/HelloREEF.cs]: fixed!

File: lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlobFileSystem.cs (hunk 3 above)
Comment (mandyshieh) [reply lang/cs/Org.Apache.REEF.IO.Tests/TestAzureBlobFileSystem.cs]: :O took it out

**Pull request merged**